src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -272,10 +272,12 @@ string EncodeUrl (Uri url)
 				if(Logger.LogNet)
 					Logger.Log (LogLevel.Info, LOG_APP, " cancelled");
 
-				httpConnection.Disconnect ();
+				await Task.Run (() => httpConnection.Disconnect ()).ConfigureAwait (false);
```

Comment:
Style: Because I hate needing to read documentation to know what `true`/`false` *means*, please use C#4 named parameters with boolean arguments:

    await Task.Run (() => httpConnection.Disconnect ())
        .ConfigureAwait (continueOnCapturedContext: false);